flang/lib/Lower/Bridge.cpp

```diff
@@ -1072,7 +1072,10 @@ class FirConverter : public Fortran::lower::AbstractConverter {
   }
 
   void genFIR(const Fortran::parser::OpenMPConstruct &omp) {
-    genOpenMPConstruct(*this, getEval(), omp);
+    auto genFIRCB = [&](Fortran::lower::pft::Evaluation &eval) {
+      genFIR(eval, /*unstructuredContext=*/false);
+    };
+    genOpenMPConstruct(*this, getEval(), omp, genFIRCB);
```

Comment:
You can also set the insertionPoint correctly in `genOpenMPConstruct` and call `genFIR` right after so you don't need to pass a callback function.

Reply:
Thanks for your suggestion! it simplified the change a lot. 👍